ui/src/app/store/zone/reducers.test.ts

```diff
@@ -1,59 +1,442 @@
+import { ZONE_ACTIONS, ZONE_PK } from "./constants";
 import reducers, { actions } from "./slice";
 
+import { ACTION_STATUS } from "app/base/constants";
 import {
   zone as zoneFactory,
+  zoneError as zoneErrorFactory,
+  zoneGenericActions as zoneGenericActionsFactory,
+  zoneModelAction as zoneModelActionFactory,
+  zoneModelActions as zoneModelActionsFactory,
   zoneState as zoneStateFactory,
 } from "testing/factories";
 
-describe("zone reducer", () => {
-  it("returns the initial state", () => {
-    expect(reducers(undefined, { type: "" })).toEqual({
-      errors: null,
-      items: [],
-      loaded: false,
-      loading: false,
-      saved: false,
-      saving: false,
+it("returns the initial state", () => {
+  expect(reducers(undefined, { type: "" })).toEqual({
+    errors: [],
+    genericActions: {
+      [ZONE_ACTIONS.create]: ACTION_STATUS.idle,
```

Comment:
I'm not sure I understand the question, but in the machine list you could add a machine and get an error, and fetch machines and get a separate error and have both visible at the same time.

Reply:
I was trying to think if we had anywhere that the same `genericAction` action could be called multiple times on the same page. e.g. if we called `model.create` multiple times in a loop we'd need to track the status for each call, even though they're the same endpoint.

Maybe something like[ this](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/bootresource/slice.ts#L76) if it was called in a loop, each call would be to the same endpoint and there's no id to key off but each call could have a different set of params.

It looks like we have this problem with `modelActions` too, e.g. the [`getLogs`](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/scriptresult/slice.ts#L208) gets [called in a loop](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsDetails/MachineTestsDetails.tsx#L51) with the same node result id but a different script type. So maybe keying of the model id isn't enough.

Reply:
This would be addressed in the follow up that tracks each individual form submission, though I suppose "form submission" might be a bit too narrow, it's more like "API call".